One observation I have is the following and I wonder if you're hitting the same thing

When i'm in a normal directory it works fine

eg. `/Users/dave/work/knative/net-istio`

```
$ ./hack/update-deps.sh
...
--- Go mod tidy
--- Go mod vendor
--- Removing unwanted vendor files
--- Removing broken symlinks
--- Checking licenses
W0414 17:10:52.528171   46112 library.go:101] "github.com/modern-go/reflect2" contains non-Go code that can't be inspected for further dependencies:
/Users/dave/work/knative/net-istio/vendor/github.com/modern-go/reflect2/relfect2_arm64.s
/Users/dave/work/knative/net-istio/vendor/github.com/modern-go/reflect2/relfect2_mips64x.s
/Users/dave/work/knative/net-istio/vendor/github.com/modern-go/reflect2/relfect2_mipsx.s
/Users/dave/work/knative/net-istio/vendor/github.com/modern-go/reflect2/relfect2_ppc64x.s
W0414 17:10:54.093586   46112 library.go:101] "golang.org/x/sys/unix" contains non-Go code that can't be inspected for further dependencies:
/Users/dave/work/knative/net-istio/vendor/golang.org/x/sys/unix/asm_bsd_arm64.s
/Users/dave/work/knative/net-istio/vendor/golang.org/x/sys/unix/zsyscall_darwin_arm64.s
W0414 17:10:57.481878   46112 library.go:101] "github.com/cespare/xxhash/v2" contains non-Go code that can't be inspected for further dependencies:
/Users/dave/work/knative/net-istio/vendor/github.com/cespare/xxhash/v2/xxhash_arm64.s
W0414 17:11:00.251015   46112 library.go:101] "github.com/prometheus/client_golang/prometheus" contains non-Go code that can't be inspected for further dependencies:
/Users/dave/work/knative/net-istio/vendor/github.com/prometheus/client_golang/prometheus/process_collector_mem_cgo_darwin.c
Deleting /var/folders/0x/lsrcyczd1yn3gggn6w9yc68m0000gn/T/tmp.I0pO64CVld
=== Removing empty go.work.sum
Deleting /var/folders/0x/lsrcyczd1yn3gggn6w9yc68m0000gn/T/tmp.I0pO64CVld
```

But if the directory path is a symlink - eg.

`/Users/dave/kn/net-istio` where kn -> points to `work/knative`

I notice I'm getting the error you mentioned

```
--- Checking licenses
F0414 17:13:21.645219   47656 main.go:77] errors for ["/Users/dave/work/knative/net-istio/..."]:
/Users/dave/work/knative/net-istio/...: -: pattern /Users/dave/work/knative/net-istio/...: directory prefix /Users/dave/work/knative/net-istio does not contain main module or its selected dependencies
exit status 1
--- FAIL: go-licenses failed the license check
Deleting /var/folders/0x/lsrcyczd1yn3gggn6w9yc68m0000gn/T/tmp.hnzdcGS0pb
Command '__go_update_deps_for_module' failed in module /Users/dave/kn/net-istio: 1
Deleting /var/folders/0x/lsrcyczd1yn3gggn6w9yc68m0000gn/T/tmp.hnzdcGS0pb
```

So I'm wondering if the issue isn't the go tool chain

--- a/library.sh
+++ b/library.sh
@@ -922,7 +922,10 @@ function run_kntest() {
 
 # Run go-licenses to check for forbidden licenses.
 function check_licenses() {
-  # Check that we don't have any forbidden licenses.
+  # Pin GOTOOLCHAIN to the project's Go version so that go-licenses is
+  # compiled with the same toolchain.  GOTOOLCHAIN=auto (the go_run default)
+  # may select a different Go, causing isStdLib() path mismatches.
+  GOTOOLCHAIN="$(go env GOVERSION)" \
   go_run github.com/google/go-licenses/v2@v2.0.1 \
     check "${REPO_ROOT_DIR}/..." || \
     { echo "--- FAIL: go-licenses failed the license check"; return 1; }

